<?xml version="1.0" encoding="UTF-8"?>
<launch> 
	<arg name="world_file"  default="/opt/ros/kinetic/share/turtlebot_gazebo/worlds/empty.world"/>
	<arg name="map_file" value="/home/moshe/catkin_ws/src/comp_cov_sim/custom_maps/map.yaml"/>
	<arg name="initial_pose_x" default="0.0"/>
	<arg name="initial_pose_y" default="0.0"/>
    	<param name="/use_sim_time" value="true" />  
  	
	<!-- Launch turtle bot world -->
	<include file="$(find comp_cov_sim)/launch/world.launch">
		<arg name="world_file" value="$(arg world_file)"/>	
	</include>

	<!-- Launch navigation stack with amcl -->
	<include file="$(find comp_cov_sim)/launch/amcl_demo.launch">
		<arg name="map_file" value="$(arg map_file)"/>
	</include>

    	<!-- Launch coverage node -->
    	<node name="coverage_node" pkg="comp_cov_sim" type="coverage_node.py" output="screen">
		<param name="robot_size" value="0.35"/>
		<param name="robot_name" value="robot_0"/>
	</node>
</launch>

